src/transactionsPage/utils.ts

```diff
@@ -30,71 +32,61 @@ export const getAppNames = (
-  return statements.reduce(
-    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-    // @ts-ignore
-    (acc: string, current: StatementStatistics, idx: number) => {
-      const newLine = idx > 0 ? "\n" : "";
-      // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-      // @ts-ignore
-      return `${acc} ${newLine} ${current.key.key_data.query}`;
-    },
-    "",
-  );
-};
+export const collectStatementsText = (statements: Statement[]): string =>
+  statements.reduce((acc: string, current: Statement, idx: number) => {
+    const newLine = idx > 0 ? "\n" : "";
+    return `${acc}${newLine}${current.key.key_data.query}`;
+  }, "");
 
 export const getStatementsById = (
   statementsIds: string[],
-  statements: StatementStatistics[],
-) => {
+  statements: Statement[],
+): Statement[] => {
   return statementsIds
     .map((id: string) => {
-      return statements.find(
-        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-        // @ts-ignore
-        (statement: StatementStatistics) => statement.id === id,
-      );
-    })
-    .filter(statement => {
-      return statement != null;
+      return statements.find((s: Statement) => s.id === id);
     })
-    .map(statement => {
-      return {
-        ...statement,
-        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-        // @ts-ignore
-        label: statement.key.key_data.query,
-      };
+    .filter((s: Statement) => {
+      return s != null;
     });
 };
 
+export const aggregateStatements = (
+  statements: Statement[],
+): AggregateStatistics[] =>
+  statements.map((s: Statement) => ({
+    label: s.key.key_data.query,
+    implicitTxn: false,
+    stats: s.stats,
+  }));
+
 export const searchTransactionsData = (
   search: string,
   transactions: Transaction[],
-  statements: StatementStatistics[],
-) => {
-  return transactions.filter((transaction: Transaction) =>
+  statements: Statement[],
+): Transaction[] => {
+  return transactions.filter((t: Transaction) =>
     search.split(" ").every(val =>
       collectStatementsText(
-        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-        // @ts-ignore
-        getStatementsById(transaction.stats_data.statement_ids, statements),
+        getStatementsById(t.stats_data.statement_ids, statements),
       )
         .toLowerCase()
         .includes(val.toLowerCase()),
     ),
   );
 };
 
-function getTimeValue(timeNumber: string, timeUnit: string) {
+function getTimeValue(timeNumber: string, timeUnit: string): number | "empty" {
   if (arguments.length < 2 || timeNumber === "0") return "empty";
   return timeUnit === "seconds"
     ? Number(timeNumber)
     : Number(timeNumber) / 1000;
 }
 
-export const filterTransactions = (data: Transaction[], filters: Filters) => {
+export const filterTransactions = (
+  data: Transaction[],
+  filters: Filters,
+): { transactions: Transaction[]; activeFilters: number } => {
```

Comment:
What is the purpose of including `activeFilters` logic in this function? I would suggest, `filterTransactions` function has to return only filtered data and have the same return type as `searchTransactionsData` function

Reply:
I was trying prevent a "Filter" component from re-rendering when calculating and storing active filters inside component's state.